server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1372,6 +1429,7 @@ public ClusterState getClusterStateUsingDiff(
             updatedIndexRouting,
             diff.isHashesOfConsistentSettingsUpdated(),
             updatedClusterStateCustom,
+            manifest.getDiffManifest() != null,
```

Comment:
This should also be `manifest.getDiffManifest().getIndicesRoutingDiffPath() != null`